src/main/webapp/app/exam/manage/exams/exam-detail.component.ts

```diff
@@ -34,16 +61,74 @@ export class ExamDetailComponent implements OnInit {
+    checkAllExamsGenerated() {
+        this.allExamsGenerated = this.exam.numberOfGeneratedStudentExams === this.exam.numberOfRegisteredUsers;
     }
 
     /**
-     * Returns the route for the student exams.
+     * Set totalPointsMandatory to true if total points of exam is smaller or equal to all mandatory points
+     * Set checkTotalPointsMandatoryOptional to true if total points of exam is bigger or equal to all mandatory points
      */
-    getStudentExamRoute() {
-        return ['/course-management', this.exam.course?.id, 'exams', this.exam.id, 'student-exams'];
+    checkTotalPointsMandatory() {
+        this.totalPointsMandatory = false;
+        this.totalPointsMandatoryOptional = false;
+        let sumPointsExerciseGroupsMandatory = 0;
+        let sumPointsExerciseGroupsOptional = 0;
+
+        // calculate mandatory points and optional points
+        if (this.pointsExercisesEqual) {
+            this.exam.exerciseGroups!.forEach((exerciseGroup) => {
+                if (exerciseGroup.isMandatory) {
+                    sumPointsExerciseGroupsMandatory += exerciseGroup!.exercises![0]!.maxPoints!;
+                } else {
+                    sumPointsExerciseGroupsOptional += exerciseGroup!.exercises![0]!.maxPoints!;
+                }
+            });
+
+            if (sumPointsExerciseGroupsMandatory <= this.exam.maxPoints!) {
+                this.totalPointsMandatory = true;
+            }
+            if (sumPointsExerciseGroupsMandatory + sumPointsExerciseGroupsOptional >= this.exam.maxPoints!) {
+                this.totalPointsMandatoryOptional = true;
+            }
+        }
+    }
+
+    /**
+     * Set pointsExercisesEqual to true if exercises have the same number of maxPoints within each exercise groups
+     */
+    checkPointsExercisesEqual() {
+        this.pointsExercisesEqual = true;
+        this.exam.exerciseGroups!.forEach((exerciseGroup) => {
+            const points = exerciseGroup.exercises?.[0].maxPoints;
+            return exerciseGroup.exercises?.forEach((exercise) => {
+                if (exercise.maxPoints !== points) {
+                    this.pointsExercisesEqual = false;
+                    return;
+                }
+            });
+        });
+    }
+
+    /**
+     * Set pointsExercisesEqual to true if exercises have the same number of maxPoints within each exercise groups
+     */
+    checkAllGroupContainsExercise() {
+        this.allGroupsContainExercise = true;
+        this.exam.exerciseGroups!.forEach((exerciseGroup) => {
+            if (!exerciseGroup.exercises || exerciseGroup.exercises.length === 0) {
+                this.allGroupsContainExercise = false;
+                return;
+            }
+        });
+    }
+
+    /**
+     * Returns the route for exam components by identifier
+     */
+    getExamRoutesByIdentifier(identifier: string) {
+        return ['/course-management', this.exam.course?.id, 'exams', this.exam.id, identifier];
```

Comment:
Would it still be a valid route or even break the app if `this.exam.course?.id` returns undefined?

Reply:
It would probably break something, as we need the course id here. I did not add this, but just refactored it, so it should be fine.